run-tests.py

```diff
@@ -68,14 +68,21 @@ def dev(args, remain):
     tag = get_tag(args.image, args.salt)
     if not image_exists(tag):
         _build(args.image, args.salt)
-    subprocess.call([
-        "docker", "run", "--rm", "-it",
+    cmd = [
+        "docker", "run", "-d",
         "-v", "{0}/test/minion.conf:/etc/salt/minion.d/minion.conf".format(BASEDIR),
         "-v", "{0}/test/salt:/srv/salt".format(BASEDIR),
         "-v", "{0}/test/pillar:/srv/pillar".format(BASEDIR),
         "-v", "{0}/{1}:/srv/formula/{1}".format(BASEDIR, _formula),
-        tag, "/bin/bash",
-    ])
+        tag,
+    ]
+    # Run the container default CMD as pid 1 (init system)
+    docker_id = subprocess.check_output(cmd).strip()
+    try:
+        # Spawn a interactive shell in the container
+        subprocess.call(["docker", "exec", "-it", docker_id, "/bin/bash"])
```

Comment:
The above command run ``/sbin/init`` doesn't give a shell (the purpose of the ``./run-tests.py dev`` command).

I splitted the concerned commit into 8621614, 253447d and 253447d hope it make things more clear and relevant to you ;)